infra/vsts_agent_windows.tf

```diff
@@ -14,12 +14,112 @@ locals {
       size       = 6,
       assignment = "default",
       disk_size  = 400,
+      pinning    = "",
     },
     {
       name       = "ci-w2"
-      size       = 0,
-      assignment = "default",
+      size       = 1,
+      assignment = "pinned",
       disk_size  = 400,
+      pinning    = <<EOF
+
+# Set workdir <> job mappings
+New-Item -ItemType Directory -Path 'D:\a'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\33'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\33\e55e0f7708a35956d0fcb3ec3f6087d9e2e1cd4b'
+Set-Content -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\33\e55e0f7708a35956d0fcb3ec3f6087d9e2e1cd4b\SourceFolder.json' `
+  -Value '{
+  "build_artifactstagingdirectory": "1\\a",
+  "agent_builddirectory": "1",
+  "collectionUrl": "https://dev.azure.com/digitalasset/",
+  "definitionName": "digital-asset.daml-daily-compat",
+  "repositoryTrackingInfo": [
+    {
+      "identifier": "self",
+      "repositoryType": "GitHub",
+      "repositoryUrl": "https://github.com/digital-asset/daml",
+      "sourcesDirectory": "1\\s\\daml"
+    }
+  ],
+  "fileFormatVersion": 3,
+  "lastRunOn": "01/26/2022 06:02:24 +00:00",
+  "repositoryType": "GitHub",
+  "lastMaintenanceAttemptedOn": "",
+  "lastMaintenanceCompletedOn": "",
+  "build_sourcesdirectory": "1\\s",
+  "common_testresultsdirectory": "1\\TestResults",
+  "collectionId": "cb88e308-485c-40f9-81b5-dcabba9e55d2",
+  "definitionId": "33",
+  "hashKey": "e55e0f7708a35956d0fcb3ec3f6087d9e2e1cd4b",
+  "repositoryUrl": "https://github.com/digital-asset/daml",
+  "system": "build"
+}'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\39'
+New-Item -ItemType Directory -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\39\55f020fa694a8855ee0847c05346f0bcf9a1308d'
+Set-Content -Path 'D:\a\SourceRootMapping\cb88e308-485c-40f9-81b5-dcabba9e55d2\39\55f020fa694a8855ee0847c05346f0bcf9a1308d\SourceFolder.json' `
+  -Value '{
+  "build_artifactstagingdirectory": "2\\a",
+  "agent_builddirectory": "2",
+  "collectionUrl": "https://dev.azure.com/digitalasset/",
+  "definitionName": "PRs",
+  "repositoryTrackingInfo": [
+    {
+      "identifier": "self",
+      "repositoryType": "GitHub",
+      "repositoryUrl": "https://github.com/digital-asset/daml",
+      "sourcesDirectory": "2\\s\\daml"
+    }
+  ],
+  "fileFormatVersion": 3,
+  "lastRunOn": "01/26/2022 15:46:13 +00:00",
+  "repositoryType": "GitHub",
+  "lastMaintenanceAttemptedOn": "",
+  "lastMaintenanceCompletedOn": "",
+  "build_sourcesdirectory": "2\\s",
```

Comment:
So if I understand this correctly you are trying to get all 3 pipelines to use different directories meaning we still won’t get caching across them but at least each pipeline will cache consistently?
Is there a chance of getting caching across pipelines?

Reply:
1. Yes, this is just about pinning each job on the same folder across all machines.
2. I don't know. It may be worth trying to set all the projects to the same folder and see what happens. I'd try that as a separate step, though, as that seems a lot harder to properly test.